qa/tasks/cephfs/fuse_mount.py

```diff
@@ -328,6 +329,7 @@ def umount(self):
         self.id = None
         self.inst = None
         self.addr = None
+        self.cleanup()
```

Comment:
Running cleanup() in umount() leads to [`run.wait([self.fuse_daemon], timeout)`](https://github.com/ceph/ceph/blob/master/qa/tasks/cephfs/fuse_mount.py#L358) in umount_wait() hang, I suppose.

I saw this entries in my logs several times -
```
2020-05-13T10:25:02.585 INFO:teuthology.orchestra.run.smithi203:> sudo bash -c 'iptables -t nat -D POSTROUTING -s 192.168.255.254/16 -o enp3s0f1 -j MASQUERADE'
2020-05-13T10:25:02.667 INFO:teuthology.orchestra.run:Running command with timeout 300
2020-05-13T10:25:02.668 INFO:teuthology.orchestra.run.smithi203:> cat /tmp/python-ceph-brx
2020-05-13T10:25:02.717 INFO:teuthology.orchestra.run.smithi203.stdout:0
2020-05-13T10:25:02.717 INFO:teuthology.orchestra.run:Running command with timeout 300
2020-05-13T10:25:02.719 INFO:teuthology.orchestra.run.smithi203:> sudo bash -c 'echo 0 > /proc/sys/net/ipv4/ip_forward'
2020-05-13T10:25:02.774 INFO:teuthology.orchestra.run:Running command with timeout 300
2020-05-13T10:25:02.775 INFO:teuthology.orchestra.run.smithi203:> rm -f /tmp/python-ceph-brx
2020-05-13T10:25:02.820 INFO:tasks.cephfs.fuse_mount:debugmsg: self.hostfs_mntpt = /home/ubuntu/cephtest/mnt.0
2020-05-13T10:25:02.821 INFO:tasks.cephfs.fuse_mount:debugmsg: self.client_id = 0
2020-05-13T10:25:02.821 INFO:tasks.cephfs.fuse_mount:debugmsg: will starting waiting now...
2020-05-13T10:25:02.822 INFO:teuthology.orchestra.run:waiting for 900
2020-05-13T10:25:20.696 INFO:teuthology.orchestra.run.smithi081:> sudo logrotate /etc/logrotate.d/ceph-test.conf
2020-05-13T10:25:20.703 INFO:teuthology.orchestra.run.smithi203:> sudo logrotate /etc/logrotate.d/ceph-test.conf
```
Here's the link to the latest one - http://qa-proxy.ceph.com/teuthology/rishabh-2020-05-13_10:05:19-fs-wip-rishabh-15070-distro-basic-smithi/5051931/teuthology.log.

I wrote this patch as a fix - https://github.com/rishabh-d-dave/ceph/commit/8555a053a634271f7f93deb4d6fcea5b59d769d7 - and reran the job which certainly went past this point - http://qa-proxy.ceph.com/teuthology/rishabh-2020-05-13_11:21:59-fs-wip-rishabh-15070-distro-basic-smithi/5051950/teuthology.log

@lxbsz Does this make sense in your opinion?

Reply:
Ran teuthology tests 4 times with improvement I mentioned in previous comment and not once did I hit the same issue. So I am moving ahead and adding that fix to this PR.

Following are links to those 4 jobs -
http://pulpito.ceph.com/rishabh-2020-05-13_14:11:11-fs-wip-rishabh-15070-distro-basic-smithi
http://pulpito.ceph.com/rishabh-2020-05-13_12:57:51-fs-wip-rishabh-15070-distro-basic-smithi
http://pulpito.ceph.com/rishabh-2020-05-13_12:37:37-fs-wip-rishabh-15070-distro-basic-smithi
http://pulpito.ceph.com/rishabh-2020-05-13_11:21:59-fs-wip-rishabh-15070-distro-basic-smithi/

Reply:
The reason is that cleanup is removing the the networking before `ceph-fuse` actually exits.